diff --git a/ajaxserver.php b/ajaxserver.php
index f35e48d..6e31f3d 100644
--- a/ajaxserver.php
+++ b/ajaxserver.php
@@ -349,7 +349,7 @@ function filedepot_dispatcher($action) {
       break;
 
     case 'delfolderperms':
-      $id = intval($_GET['id']);
+      $id = intval($_POST['id']);
       if ($id > 0) {
         $query = db_query("SELECT catid, permtype, permid FROM  {filedepot_access} WHERE accid=:accid", array(
           ':accid' => $id,
diff --git a/js/initapplication.js b/js/initapplication.js
index bc2e4ab..c6a2e49 100644
--- a/js/initapplication.js
+++ b/js/initapplication.js
@@ -217,7 +217,8 @@ var init_filedepot = function() {
       Event.preventDefault(e);
       var params = target.getAttribute('href');
       var id = parseURL(params,'accid');
-      var surl = ajax_post_handler_url + '/delfolderperms&id=' + id;
+      var surl = ajax_post_handler_url + '/delfolderperms';
+      var postdata = '&id=' + id;
       var callback = {
         success: function(o) {
           var json = o.responseText.substring(o.responseText.indexOf('{'), o.responseText.lastIndexOf('}') + 1);
@@ -238,7 +239,7 @@ var init_filedepot = function() {
         timeout:55000
       }
       updateAjaxStatus(NEXLANG_activitymsg4);
-      YAHOO.util.Connect.asyncRequest('POST', surl, callback);
+      YAHOO.util.Connect.asyncRequest('POST', surl, callback, postdata);
     }
   });
 
